geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -539,14 +532,12 @@ protected boolean processHandShake() {
         return false;
       }
       synchronized (getCleanupProxyIdTable()) {
-        Counter numRefs = (Counter) getCleanupProxyIdTable().get(this.proxyId);
+        Integer numRefs = getCleanupProxyIdTable().get(this.proxyId);
         if (numRefs != null) {
-          numRefs.incr();
+          ++numRefs;
```

Comment:
same problem here - incrementing an integer isn't going to update the map

Reply:
I'll add some checks in the integration test to verify that we've properly incremented this state.

Reply:
I wrote a simple test that showed the value in the map is unchanged.  With a Counter the value in the map would be incremented but it's not going to do that with a boxed integer.

map.put(key, 1);
Integer anInt = map.get(key);
anInt++;
Integer anotherInt = map.get(key);
System.out.println("anInt="+anInt+"; value in map="+anotherInt);

The result will be 2 and 1.

Reply:
Thanks Bruce, I'm an idiot and managed to be testing the one case that wasn't actually incrementing the value (going from 0 to 1 and back).  Reverted this to use the counter again, which I'm not thrilled about, but seems less error prone than the other solutions I looked at.

Reply:
Would it be clearer to rename it `MutableInteger`?